hunsri/EwMM-ProjectCS
# feat: settings

## Description
Adjust SFX volume, music volume and NPC behaviors based on difficulty chosen in the main menu.

## How to Test
- Set your own volume and difficulty level on `MainMenu` scene
- Play basically any scene that has `SoundManager` and `NPCSpawner` in it
- Notice how the NPC effect sounds (cough, sneeze) changes -> Is a little bit difficult sometime to notice the difference, as you have to be close to the NPC because the audio comes from a point in the 3D world.
- Try to set the volume to 0 -> NPC won't play any SFX on cough / sneeze
- Changing difficulty would result the following changes (could be set on `NPCDifficultyLevel.cs`):
  - Speed (infected, cured, healthy)
  - Infection rate

## TODO
- [x] Adjust music (?) based on settings slider on main menu

*This branch is checked out from the `attach-mask-onhit` #34 branch. So this will still be WIP until it is merged.*
